csrc/quantization/fp8/common.cu

```diff
@@ -124,6 +130,34 @@ __global__ void scaled_fp8_quant_kernel(c10::Float8_e4m3fn* __restrict__ out,
   }
 }
 
+template <typename Tout, typename Tin, int Vec_size>
+__global__ void convert_fp8_kernel(const Tin* __restrict__ src_data,
+                                   Tout* __restrict__ dst_data,
+                                   const float* scale, size_t N) {
+  using V_in_vec = typename Vec<Tin, Vec_size>::Type;
+  using V_out_vec = typename Vec<Tout, Vec_size>::Type;
+  auto dst_data_vec = reinterpret_cast<V_out_vec*>(dst_data);
+  auto src_data_vec = reinterpret_cast<const V_in_vec*>(src_data);
+
+  int64_t startIdx = (threadIdx.x + blockDim.x * blockIdx.x);
+  auto idx = startIdx;
+  if (idx >= N) {
+    return;
+  }
+#if defined(ENABLE_FP8)
+  #ifdef USE_ROCM
+  dst_data_vec[idx] = fp8::scaled_vec_conversion<V_out_vec, V_in_vec>(
+      src_data_vec[idx], *scale);
+  #else
+  dst_data_vec[idx] =
+      fp8::scaled_convert<V_out_vec, V_in_vec, Fp8KVCacheDataType::kFp8E4M3>(
+          src_data_vec[idx], *scale);
+  #endif
```

Comment:
Can we unify the interface and avoid introduce `#ifdef USE_ROCM` here? We should minimize such branches as possible.